tools/issue-labeler/src/AzureRAGService/TriageRAG.cs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Search.Documents;
+using Microsoft.Extensions.Logging;
+using Azure.AI.OpenAI;
+using Azure.Search.Documents.Models;
+using OpenAI.Chat;
+using Azure.Identity;
+using System.Text.Json;
+
+namespace AzureRAGService
+{
+    public interface ITriageRAG
+    {
+        IEnumerable<(T, double)> AzureSearchQuery<T>(Uri searchEndpoint, string indexName, string semanticConfigName, string field, DefaultAzureCredential credential, string query, int count);
+        string SendMessageQna(Uri openAIEndpoint, DefaultAzureCredential credential, string modelName, string instructions, string message, BinaryData structure);
+    }
+
+    public class TriageRAG : ITriageRAG
+    {
+        private ILogger<ITriageRAG> _logger;
+
+        public TriageRAG(ILogger<ITriageRAG> logger)
+        {
+            _logger = logger;
+        }
+
+
+        /// <summary>
+        /// Executes an Azure Search query.
+        /// </summary>
+        /// <typeparam name="T">The type of the search result.</typeparam>
+        /// <param name="searchEndpoint">The search endpoint URI.</param>
+        /// <param name="indexName">The name of the search index.</param>
+        /// <param name="semanticConfigName">The name of the semantic configuration.</param>
+        /// <param name="field">The field to search.</param>
+        /// <param name="credential">The Azure credential.</param>
+        /// <param name="query">The search query.</param>
+        /// <param name="count">The number of results to return.</param>
+        /// <returns>An enumerable of "search results" with their associated scores.</returns>
+        public IEnumerable<(T, double)> AzureSearchQuery<T>(
```

Comment:
Given that we're making a service call and are I/O bound, this should probably be using the async form.